.github/workflows/requirements.yml

```diff
@@ -0,0 +1,119 @@
+        run: |
+          pip install -r dev/requirements.txt
+      - uses: ./.github/actions/update-requirements
+      - name: Install dependencies
+        run: |
+          source ./dev/install-common-deps.sh --skinny
+      - name: Run tests
+        run: |
+          ./dev/run-python-skinny-tests.sh
+
+  core:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - uses: ./.github/actions/setup-python
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - uses: ./.github/actions/setup-pyenv
+      - uses: ./.github/actions/setup-java
+      - name: Install dev script dependencies
+        run: |
+          pip install -r dev/requirements.txt
+      - uses: ./.github/actions/update-requirements
+      - name: Install dependencies
+        run: |
+          source ./dev/install-common-deps.sh --ml
+      - name: Run tests
+        run: |
+          source dev/setup-ssh.sh
+          ./dev/run-python-tests.sh
+
+  create-pr:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'schedule'
+    needs: [skinny, core]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - uses: ./.github/actions/setup-python
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Install dev script dependencies
+        run: |
+          pip install -r dev/requirements.txt
+      - uses: ./.github/actions/update-requirements
+      - name: Create PR branch name
+        id: pr-branch
+        run: |
+          echo "::set-output name=name::update-requirements-$(git diff | sha256sum | head -c 8)"
+      - uses: actions/github-script@v6
+        id: pr-exists
+        with:
+          result-encoding: string
+          script: |
+            const recentPrs = await github.rest.pulls.list({
+              owner: 'mlflow',
+              repo: 'mlflow',
+              per_page: 100,
+              sort: 'created',
+              direction: 'desc',
+            });
+            const exists = recentPrs.data.some(
+              pr => pr.head.ref === '${{ steps.pr-branch.outputs.name }}'
+            );
+            return exists.toString();
+      # If there are no changes, the create-pull-request action exits silently
+      - uses: peter-evans/create-pull-request@v4
+        if: steps.pr-exists.outputs.result != 'true'
+        with:
+          token: ${{ secrets.MLFLOW_AUTOMATION_TOKEN }}
+          add-paths: requirements
+          commit-message: 'Update requirements'
+          signoff: true
+          committer: 'mlflow-automation <mlflow-automation@users.noreply.github.com>'
+          author: 'mlflow-automation <mlflow-automation@users.noreply.github.com>'
+          branch: ${{ steps.pr-branch.outputs.name }}
+          title: 'Update requirements'
+          body: |
+            Build Log: https://github.com/mlflow/mlflow/actions/runs/${{ github.run_id }}
```

Comment:
Is the idea that we'll have our MLflow release pipeline trigger this GitHub Actions workflow?

Reply:
The release pipeline will run `dev/update_requirements.py` with `--check` flag to ensure `core-requirements.yaml` and `skinny-requirements.yaml` are up to date.

Reply:
This workflow daily checks if there are updated packages and files a PR.